pkg/controller/hyperconverged/hyperconverged_controller_test.go

```diff
@@ -2,230 +2,993 @@ package hyperconverged
+						Name:      name,
+						Namespace: namespace,
+					},
+					Spec: hcov1alpha1.HyperConvergedSpec{},
+				}
+
+				expectedResource := newKubeVirtForCR(hco, namespace)
+				cl := initClient([]runtime.Object{})
+				r := initReconciler(cl)
+				Expect(r.ensureKubeVirt(hco, log, request)).To(BeNil())
+
+				foundResource := &kubevirtv1.KubeVirt{}
+				Expect(
+					cl.Get(context.TODO(),
+						types.NamespacedName{Name: expectedResource.Name, Namespace: expectedResource.Namespace},
+						foundResource),
+				).To(BeNil())
+				Expect(foundResource.Name).To(Equal(expectedResource.Name))
+				Expect(foundResource.Labels).Should(HaveKeyWithValue("app", name))
+				Expect(foundResource.Namespace).To(Equal(expectedResource.Namespace))
 			})
-		})
 
-		Context("Network Addons CR", func() {
-			It("should have metadata and spec and namespace should be unspecified", func() {
-				cr := newNetworkAddonsForCR(instance, "")
-				checkMetadata(cr.ObjectMeta, networkaddonsnames.OPERATOR_CONFIG, appLabel, "")
-				Expect(cr.Spec.Multus).To(Equal(&networkaddons.Multus{}))
-				Expect(cr.Spec.LinuxBridge).To(Equal(&networkaddons.LinuxBridge{}))
-				Expect(cr.Spec.KubeMacPool).To(Equal(&networkaddons.KubeMacPool{}))
+			It("should find if present", func() {
+				hco := &hcov1alpha1.HyperConverged{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      name,
+						Namespace: namespace,
+					},
+					Spec: hcov1alpha1.HyperConvergedSpec{},
+				}
+
+				expectedResource := newKubeVirtForCR(hco, namespace)
+				expectedResource.ObjectMeta.SelfLink = fmt.Sprintf("/apis/v1/namespaces/%s/dummies/%s", expectedResource.Namespace, expectedResource.Name)
+				cl := initClient([]runtime.Object{hco, expectedResource})
+				r := initReconciler(cl)
+				Expect(r.ensureKubeVirt(hco, log, request)).To(BeNil())
+
+				// Check HCO's status
+				Expect(hco.Status.RelatedObjects).To(Not(BeNil()))
+				objectRef, err := reference.GetReference(r.scheme, expectedResource)
+				Expect(err).To(BeNil())
+				// ObjectReference should have been added
+				Expect(hco.Status.RelatedObjects).To(ContainElement(*objectRef))
+				// Check conditions
+				Expect(r.conditions).To(ContainElement(RepresentCondition(conditionsv1.Condition{
+					Type:    conditionsv1.ConditionAvailable,
+					Status:  corev1.ConditionFalse,
+					Reason:  "KubeVirtConditions",
+					Message: "KubeVirt resource has no conditions",
+				})))
+				Expect(r.conditions).To(ContainElement(RepresentCondition(conditionsv1.Condition{
+					Type:    conditionsv1.ConditionProgressing,
+					Status:  corev1.ConditionTrue,
+					Reason:  "KubeVirtConditions",
+					Message: "KubeVirt resource has no conditions",
+				})))
+				Expect(r.conditions).To(ContainElement(RepresentCondition(conditionsv1.Condition{
+					Type:    conditionsv1.ConditionUpgradeable,
+					Status:  corev1.ConditionFalse,
+					Reason:  "KubeVirtConditions",
+					Message: "KubeVirt resource has no conditions",
+				})))
```

Comment:
Why isn't conditionsv1.ConditionDegraded checked here? I see it used in line 227.

Reply:
We don't set the degraded condition when the kubevirt resource has no conditions. On 227 I had specifically crafted a KubeVirt resource with the degraded condition set to true.